audbackend/core/backend/base.py

```diff
@@ -781,6 +817,7 @@ def open(
 
         """
         utils.call_function_on_backend(self._open)
+        self.opened = True
```

Comment:
I wonder if we should rather add a return value to `self._open()` and assign the result here. This avoids that we set the flag when actually no connection could be established.

Reply:
I see the point, but I'm not sure if it will work out in reality.
Inside `self._open()` it can happen that an error is raised when no connection can be established. If everything went fine, it will return `True` at the end, but when will it ever return `False`?

Reply:
Instead of raising an error, the backend could return `False` if no connection could be established.

Reply:
But when not raising an error, the user has no clue what goes wrong.

Reply:
We could raise an error in the base class if we want to. But I guess you will argument that it's better if the backend raises the error to show a more specific error message. So maybe we should then document somewhere that an error should be raised if the implementation cannot establish a connection or close it.

Reply:
I adjusted the docstrings to mention that an error should be raised:

```python
    def _close(
        self,
    ):  # pragma: no cover
        r"""Close connection to repository.

        An error should be raised,
        if the connection to the backend
        cannot be closed.

        """
        pass

# ...

    def _open(
        self,
    ):  # pragma: no cover
        r"""Open connection to backend.

        If repository does not exist,
        or the backend cannot be opened,
        an error should be raised.

        """
        pass
```